integration/graphql/src/main/java/com/blazebit/persistence/integration/graphql/GraphQLEntityViewSupport.java

```diff
@@ -624,50 +602,41 @@ public void applyFetches(DataFetchingEnvironment dataFetchingEnvironment, Entity
+            List<String> mappedFields = new ArrayList<>();
+            for (int i = 0; i < fieldParts.length; i++) {
+                String[] fieldPartComponents = fieldParts[i].split("\\.");
+                String typeName = fieldPartComponents[0];
+                String fieldName = fieldPartComponents[1];
+                if (elementRootParts.length > i && elementRootParts[i].equals(fieldName)) {
+                    // ignore the element root prefix
                     continue;
                 }
-                GraphQLFieldsContainer baseType = rootType;
-                sb.setLength(0);
-                int fieldStartIndex = 0;
-                for (int i = 0; i < key.length(); i++) {
-                    final char c = key.charAt(i);
-                    if (i < prefix.length()) {
-                        if (c != prefix.charAt(i)) {
-                            continue OUTER;
-                        } else {
-                            continue;
-                        }
-                    }
-                    if (c == '/') {
-                        if ((baseType = (GraphQLFieldsContainer) applyFieldMapping(sb, baseType, fieldStartIndex)) == null) {
-                            continue OUTER;
-                        }
-                        sb.append('.');
-                        fieldStartIndex = sb.length();
-                    } else {
-                        sb.append(c);
-                    }
+                if (META_FIELDS.contains(fieldName)) {
+                    // If this is a GraphQL meta field (e.g. __typename) then there's no EV mapping for it but still
+                    // we must fetch info from DB. So put the intermediate field path into fetches, even if it leads
+                    // into fetching the whole subtree in EntityViewConfiguration#getFetches (FIXME)
+                    continue;
                 }
-                if (sb.length() > 0 && !META_FIELDS.contains(sb.substring(fieldStartIndex))) {
-                    // Include only actual leaf types into setting.fetches because intermediate objects types
-                    // lead to fetching the whole subtree in EntityViewConfiguration#getFetches
-                    GraphQLType fieldType = applyFieldMapping(sb, baseType, fieldStartIndex);
-                    if (fieldType != null && !(fieldType instanceof GraphQLCompositeType)) {
-                        setting.fetch(sb.toString());
-                    }
+                String mappedFieldPart = applyFieldMapping(typeName, fieldName);
+                if (mappedFieldPart == null) {
+                    // fieldName cannot be mapped to an entity view field -> ignore the whole field
+                    continue OUTER;
                 }
+                mappedFields.add(mappedFieldPart);
+            }
+            if (!mappedFields.isEmpty()) {
+                setting.fetch(String.join(".", mappedFields));
             }
```

Comment:
Would be very happy if you could append to a `StringBuilder` directly instead of building an intermediate list first :)
Also, the `StringBuilder` can be reused by calling `setLength(0)` at the start of the loop :)

Reply:
Again, did you test this? From what I read (e.g. https://stackoverflow.com/questions/585860/string-join-vs-stringbuilder-which-is-faster) the `String.join` will not be significantly slower and might be actually faster in our case as it should be quite optimized.

Reply:
It's not about being faster, it's about the allocation rate due to the other unnecessary objects, `ArrayList` and the containing `Object[]`. Possible `byte[]` resizes etc.